### DISCUSSION TITLE
Adding new quotes

### DISCUSSION THREAD

**Discussion by lukePeavey** [lukePeavey/quotable#69] [Quotable Data]
## Update

**The source data for the API is now stored in a public repo on Github**.   If you would like contribute new quotes or help improve the existing content, please visit the data repo: https://github.com/quotable-io/data

**Comment by lukePeavey** [comment]
## Example

I would like to add the following quotes

```json
[
  {
    "content": "I'll prepare and some day my chance will come.",
    "author": "Abraham Lincoln",
    "tags": ["famous-quotes", "inspirational"]
  },
  {
    "content": "Action is eloquence!",
    "author": "William Shakespeare",
    "tags": ["literature", "famous-quotes"]
  }
]
```

**Comment by vpodolyan** [comment]
I'd like to add these quotes. Maybe some are already in the database

```
[
  {
    "content": "The biggest room in the world is room for improvement.",
    "author": "Helmut Schmidt",
    "tags": ["famous-quotes", "inspirational"]
  },
  {
    "content": "There’s power in looking silly and not caring that you do.",
    "author": "Amy Poehler",
    "tags": ["famous-quotes", "inspirational"]
  },
  {
    "content": "In times of change, learners inherit the earth, while the learned find themselves beautifully equipped to deal with a world that no longer exists.",
    "author": "Eric Hoffer",
    "tags": ["famous-quotes", "inspirational"]
  },
  {
    "content": "Whether you think you can or you think you can't, you are right.",
    "author": "Henry Ford",
    "tags": ["famous-quotes", "inspirational"]
  },
  {
    "content": "If you're changing the world, you're working on important things. You're excited to get up in the morning.",
    "author": "Larry Page",
    "tags": ["famous-quotes", "inspirational"]
  },
    {
    "content": "The one thing that you have that nobody else has is you. Your voice, your mind, your story, your vision. So write and draw and build and play and dance and live as only you can.",
    "author": "Neil Gaiman",
    "tags": ["famous-quotes", "inspirational"]
  },
]
```

**Reply by lukePeavey** [reply]
Thanks, these are great.  I'll add them tomorrow!

**Reply by vpodolyan** [reply]
Perfect, glad it's useful!

**Reply by lukePeavey** [reply]
I added your quotes to database. 

See latest quotes

[/quotes?sortBy=dateAdded](https://api.quotable.io/quotes?sortBy=dateAdded)

See latest authors

[/authors?sortBy=dateAdded](https://api.quotable.io/authors?sortBy=dateAdded)

**Reply by vpodolyan** [reply]
Oh, thanks a lot! Sorry, looks like I've missed notification about your comment :slightly_smiling_face:

**Comment by sud002** [comment]


**Reply by lukePeavey** [reply]
Thanks, I'll take a look at these

**Comment by Pia007** [comment]
Hi, I would like to suggest the following quotes if they have not been added already.

    [
        {
          "content": "The real power belongs to the people.",
          "author": "Greta Thunberg",
          "tags": ["famous-quotes", "inspirational"]
        },
        {
          "content": "But it is the sufferings of the many which pay for the luxuries of a few",
          "author": "Greta Thunberg",
          "tags": ["famous-quotes", "climate-change"]
        },
        {
         "content": "So let us wage a global struggle against illiteracy, poverty and terrorism, and let us pick up our books and pens. They are our most powerful weapons.",
         "author": "Malala Yousafzai",
         "tags": ["famous-quotes", "inspirational"]
        },
        {
         "content": "One child, one teacher,  one pen, and one book can change the world.",
         "author": "Malala Yousafzai",
         "tags": ["famous-quotes", "inspirational"]
        },
        {
         "content": "Don't be part of the spectacle; be truthful about what you care about and speak from the heart.",
         "author": "David Hogg",
         "tags": ["famous-quotes", "honesty"]
        },
        {
         "content": "In the same way that we all have the right to bear arms, we all have a right to live.",
         "author": "David Hogg",
         "tags": ["famous-quotes", "rights"]
        },
        {
         "content": "If you are willing to fail interestingly, you will succeed interestingly.",
         "author": "Edward Albee",
         "tags": ["famous-quotes", "inspirational"]
        },
        {
         "content": "Remember one thing about democracy. We can have anything that we want and at the same time, we always get what we deserve.",
         "author": "Edward Albee",
         "tags": ["famous-quotes", "democracy"]
        },
        {
         "content": "I never really had a classic struggle. I had faith.",
         "author": "Denzel Washington",
         "tags": ["famous-quotes", "faith"]
        },
        {
         "content": "Acting is just a way of making a living. Family is life.",
         "author": "Denzel Washington",
         "tags": ["famous-quotes", "family"]
        },
        {
         "content": "You are indebted to life until you help some less fortunate person, just as you were helped.",
         "author": "Melinda Gates",
         "tags": ["inspiring-quotes", "philanthropy"]
        },
        {
         "content": "Connect deeply with others. Our humanity is the only thing that we all have in common.",
         "author": "Melinda Gates",
         "tags": ["inspiring-quotes", "humanity"]
        }
    ]

**Reply by lukePeavey** [reply]
thanks these look good. I’ll add them to the collection

**Reply by Pia007** [reply]
Awesome! Thanks!

**Comment by AngelliaX** [comment]
hi, is there any way to see a full list of all quotes already included in the database

the closest way I can find is to use this URL, which only allows up to 150:
https://api.quotable.io/quotes?limit=150

, but so far there are more than 1885 quotes now

I want to add some quotes but I'm afraid it's already inside the database, and the only way to check is to use this URL https://api.quotable.io/search/quotes?query=" "
but it is way more convenient if having full 1885 quotes displayed on the browser and check with ctrl+f

but it already has almost 2000 quotes, so i don't think it will be necessary to add more

**Reply by lukePeavey** [reply]
The source data for the API is now in a public repository on Github. Hopefully this will make it easier for others to contribute to the collection of quotes. 

https://github.com/quotable-io/data

**Reply by lukePeavey** [reply]
If you would like to add new quotes, please submit them on the data repo.

**Reply by OfficialPhilcomm** [reply]
I'd like to submit some quotes, but I am confused about the `_id`, `authorId` and `authorSlug` fields. Could you explain how these are generated?

Great project btw, and amazing that you made it freely accessible

**Reply by lukePeavey** [reply]
> I'd like to submit some quotes, but I am confused about the `_id`, `authorId` and `authorSlug` fields. Could you explain how these are generated?
> 
> 
> 
> Great project btw, and amazing that you made it freely accessible

Check out the data repository. It includes documentation on adding new quotes...

we use A CLI script to add new quotes. You provide a JSON array of quote objects, each with the following fields: content, author (full name), and tags (optional).  the scripts filters out any duplicate quotes that already in the collection, then creates the necessary Quote, Author, and Tag objects and adds them their respective collections.

So all fields you mentioned are generated automatically.

If you are interested, the source code for all CLI scripts is in the data repo.

**Reply by lukePeavey** [reply]
If you want to submit new quotes, you can open an issue on the data repository to suggest new quotes. 

Check out the contributors guide on the data repo for more details. 

Https://github.com/quotable-io/data

**Comment by sad-adnan** [comment]
The Source repo has a lot of tags but the api shows only 20 tags, I'm confused , am i missing anything? @lukePeavey  can you please clear this or if you could add new tags and quotes to the api would be helpful ,  Thanks in advance. 

data repo tag link :  https://raw.githubusercontent.com/quotable-io/data/master/data/tags.json
tag api link tha i'm using :  https://api.quotable.io/tags

**Comment by lukePeavey** [comment]
Those changes just haven't been synced to the MongoDB database yet.

**Comment by lukePeavey** [comment]
@sad-adnan Just pushed the latest changes to MongoDB.

The [list of tags](https://api.quotable.io/tags) matches the data in the repo now.

**Reply by sad-adnan** [reply]
@lukePeavey Thank you very much.

**Comment by pleasedonotdisturb** [comment]
Hello! I would like to add these quotes. Just in case they weren’t added :)
```
[
  {
    “content”: “ That person who helps others simply because it should or must be done, and because it is the right thing to do, is indeed without a doubt, a real superhero.”,
    “author”: “Stan Lee”,
    “tags”: ["character”]
  },
  {
    “content”: “Love is so short, but forgetting is so long.”,
    “author”: “Pablo Neruda”,
    “tags”: ["love”]
  }
]
```
P.S. Does the database allow bible quotes?

**Comment by adisonmasih** [comment]
Hey! i would like to add these ones!
```json
[
  {
    "content": "Too much or too little, neither brings joy. Enough does.",
    "author": "Adison Masih",
    "tags": ["happiness"]
  },
 {
    "content": "Testing people before you need them can prevent unpleasant surprises later.",
    "author": "Adison Masih",
    "tags": ["friendship", "politics", "business"]
  }
]
```
